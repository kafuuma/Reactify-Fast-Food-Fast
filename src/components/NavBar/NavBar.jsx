import React from 'react';
import {
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavbarToggler,
  MDBCollapse,
  MDBNavItem,
  MDBNavLink
} from 'mdbreact';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      collapse: false,
      isWideEnough: false,
      homeActive: false,
      signupActive: false,
      loginActive: false
    };
    false;
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.setState({
      collapse: !this.state.collapse
    });
  }

  homeActive = () => {
    this.setState({
      homeActive: true,
      signupActive: false,
      loginActive: false
    });
  };

  signupActive = () => {
    this.setState({
      homeActive: false,
      signupActive: true,
      loginActive: false
    });
  };
  loginActive = () => {
    this.setState({
      homeActive: false,
      signupActive: false,
      loginActive: true
    });
  };

  render() {
    const bgPink = { backgroundColor: '#ff3300' };
    return (
      <div>
        <header>
          <MDBNavbar style={bgPink} dark expand="md" fixed="top">
            <MDBNavbarBrand href="/">
              <strong>Fast Food Fast</strong>
            </MDBNavbarBrand>
            {!this.state.isWideEnough && (
              <MDBNavbarToggler id="togler" onClick={this.onClick} />
            )}
            <MDBCollapse isOpen={this.state.collapse} navbar>
              <MDBNavbarNav right>
                <MDBNavItem
                  onClick={this.homeActive}
                  active={this.state.homeActive}
                >
                  <MDBNavLink to="/">Home</MDBNavLink>
                </MDBNavItem>
                <MDBNavItem
                  onClick={this.signupActive}
                  active={this.state.signupActive}
                >
                  <MDBNavLink to="/signup">Signup</MDBNavLink>
                </MDBNavItem>
                <MDBNavItem
                  onClick={this.loginActive}
                  active={this.state.loginActive}
                >
                  <MDBNavLink to="/login">Login</MDBNavLink>
                </MDBNavItem>
              </MDBNavbarNav>
            </MDBCollapse>
          </MDBNavbar>
        </header>
      </div>
    );
  }
}

export default NavBar;
