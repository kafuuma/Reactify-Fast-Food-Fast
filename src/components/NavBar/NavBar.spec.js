import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';
const onClick = jest.fn;

describe('<NaBbar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  it('matches snapshort', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should should simulate click', () => {
    wrapper.find('#togler').simulate('click');
    expect(onClick()).toBeDefined;
  });
  it('should call homeActive ', () => {
    const instance = wrapper.instance();
    expect(
      instance.homeActive({
        preventDefault: jest.fn,
        homeActive: true,
        signupActive: false,
        loginActive: false
      })
    );
  });

  it('should call signupActive ', () => {
    const instance = wrapper.instance();
    expect(
      instance.signupActive({
        preventDefault: jest.fn,
        homeActive: false,
        signupActive: true,
        loginActive: false
      })
    );
  });

  it('should call loginActive ', () => {
    const instance = wrapper.instance();
    expect(
      instance.loginActive({
        preventDefault: jest.fn,
        homeActive: false,
        signupActive: false,
        loginActive: true
      })
    );
  });
});
