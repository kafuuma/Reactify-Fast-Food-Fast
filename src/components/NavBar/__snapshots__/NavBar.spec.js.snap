// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NaBbar /> matches snapshort 1`] = `ShallowWrapper {}`;
